pkg/generator/specification.go

```diff
@@ -118,51 +126,50 @@ func (s *SpecificationGenerator) generateIATPerGranularity(numberOfInvocations i
 		finalIAT := append([]float64{beginningIAT}, iatResult[i+1:]...)
 		finalIAT = append(finalIAT, iatResult[:i]...)
 		iatResult = append(finalIAT, endIAT)
-	} else {
-		iatResult = append([]float64{0.0}, iatResult...)
 	}
 
 	return iatResult, totalDuration
 }
 
 // GenerateIAT generates IAT according to the given distribution. Number of minutes is the length of invocationsPerMinute array
-func (s *SpecificationGenerator) generateIAT(invocationsPerMinute []int, iatDistribution common.IatDistribution, shiftIAT bool, granularity common.TraceGranularity) (common.IATMatrix, common.ProbabilisticDuration) {
-	var IAT [][]float64
+func (s *SpecificationGenerator) generateIAT(invocationsPerMinute []int, iatDistribution common.IatDistribution,
+	shiftIAT bool, granularity common.TraceGranularity) (common.IATArray, []int, common.ProbabilisticDuration) {
+
+	var IAT []float64
+	var perMinuteCount []int
 	var nonScaledDuration []float64
 
 	numberOfMinutes := len(invocationsPerMinute)
 	for i := 0; i < numberOfMinutes; i++ {
-		minuteIAT, duration := s.generateIATPerGranularity(invocationsPerMinute[i], iatDistribution, shiftIAT, granularity)
+		minuteIAT, duration := s.generateIATPerGranularity(i, invocationsPerMinute[i], iatDistribution, shiftIAT, granularity)
```

Comment:
When we have 0 invocations per minute, we return the array with a single 0. It doesn't seem right to append this to IAT array. There should be IAT > 60s, instead.

Reply:
Fixed in https://github.com/vhive-serverless/invitro/pull/557/commits/6d29c101488f90ffdb08adcbda36cb7d06987ca7.